src/AutoRest.CSharp.V3/Generation/Writers/RestClientWriter.cs

```diff
@@ -136,31 +136,32 @@ private void WriteRequestCreation(CodeWriter writer, RestClientMethod operation)
                     WriteHeader(writer, header);
                 }
 
-                if (operation.Request.Body is RequestBody body && body.Serialization is JsonSerialization jsonSerialization)
+                if (operation.Request.Body is SchemaRequestBody body)
                 {
-                    writer.Line($"using var content = new {typeof(Utf8JsonRequestContent)}();");
-
                     ParameterOrConstant value = body.Value;
-
-                    writer.ToSerializeCall(
-                        jsonSerialization,
-                        WriteConstantOrParameter(value, ignoreNullability: true),
-                        writerName: w => w.Append($"content.{nameof(Utf8JsonRequestContent.JsonWriter)}"));
-
-                    writer.Line($"request.Content = content;");
+                    switch (body.Serialization)
+                    {
+                        case JsonSerialization jsonSerialization:
+                            writer.Line($"using var content = new {typeof(Utf8JsonRequestContent)}();");
+                            writer.ToSerializeCall(
+                                jsonSerialization,
+                                WriteConstantOrParameter(value, ignoreNullability: true),
+                                writerName: w => w.Append($"content.{nameof(Utf8JsonRequestContent.JsonWriter)}"));
+                            writer.Line($"request.Content = content;");
+                            break;
+                        case XmlElementSerialization xmlSerialization:
+                            writer.Line($"using var content = new {typeof(XmlWriterContent)}();");
+                            writer.ToSerializeCall(
+                                xmlSerialization,
+                                WriteConstantOrParameter(value, ignoreNullability: true),
+                                writerName: w => w.Append($"content.{nameof(XmlWriterContent.XmlWriter)}"));
+                            writer.Line($"request.Content = content;");
+                            break;
```

Comment:
default throw?

Reply:
The original code didn't throw here. This switch is stylistically the same as the rest of the serialization handling. It is also equivalent to the new WriteDeserializationForMethods on CodeWriterExtensions (your suggestion to encapsulate the switch itself) which also doesn't throw.

Either we should throw whenever we can't handle something (which we aren't particularly consistent on) or we don't, and let the code write gracefully (meaning, it still writes compile-able code, but just doesn't have that section). In this case here, if we don't write this section (switch falls through), it will still write a valid method. The request simply won't have content.

Reply:
I think it might be time to tighten things down now that we have most of the features. This would allow us to easily discover corner cases.

Reply:
I added throws in a similar way that I found elsewhere (NotImplementedException) to any relevant serialzation processing switches.